### ISSUE TITLE
debug help: training with pre-calculated phoneme durations

### ISSUE THREAD

**Issue by dbkest** [shivammehta25/Matcha-TTS#106]
Hello, I would like to ask a few questions about my experiment. I followed the training steps for the LJspeech dataset exactly, with the only modification being replacing LJspeech with my English data. This allowed me to generate high-quality speech. Building on this, I conducted another experiment where I used pre-calculated phoneme durations instead of using MAS to obtain the phoneme durations. I found that the generated audio spectrograms were good, with clear harmonic structures, but it was difficult to understand what was being said. Could you give me some debugging directions? Have you conducted this experiment before? I performed a few troubleshooting steps:

I used the same data with FastSpeech2, ruling out alignment issues. I also printed the data loaded by the Matcha-TTS dataloader, confirming that the phonemes and durations were aligned.
Compared to FastSpeech2, the main difference in Matcha-TTS, aside from the decoder, is that the latter uses a prior loss. I am currently retraining with this loss turned off, but it has only just begun.

Thank you very much!

generated audio with model which trained with pre-calculated phoneme durations
text: You're a chef. You're a young chef, yes?
<img width="899" alt="截屏2024-10-18 01 01 34" src="https://github.com/user-attachments/assets/ecc6118a-63d1-4b00-b205-5a9e980f89c0">

The loss curves from the experiment are as follows, with the blue line representing the experiment using MAS and the gray line representing the experiment using pre-calculated durations.

<img width="849" alt="截屏2024-10-18 00 50 03" src="https://github.com/user-attachments/assets/6b5c53c4-a146-4296-8b1d-2f6c9e2e9fdb">
<img width="858" alt="截屏2024-10-18 00 50 29" src="https://github.com/user-attachments/assets/53f7a1fc-43f0-4b57-8938-6f39f081b810">
<img width="855" alt="截屏2024-10-18 00 50 56" src="https://github.com/user-attachments/assets/96c41633-706d-45d7-a020-0ecb71a5409e">
<img width="858" alt="截屏2024-10-18 00 51 24" src="https://github.com/user-attachments/assets/fbc82c89-2e4a-40dc-af25-91467dfe9c98">
<img width="857" alt="截屏2024-10-18 00 51 47" src="https://github.com/user-attachments/assets/78409f9f-84b8-4c59-b1c0-ddddde30a4ef">
<img width="850" alt="截屏2024-10-18 00 52 02" src="https://github.com/user-attachments/assets/97b2d8d0-a0ff-4039-9c0a-eba19b85c9fa">
<img width="854" alt="截屏2024-10-18 00 52 19" src="https://github.com/user-attachments/assets/77a0a440-c2dd-4847-a25f-03eceed533db">
<img width="857" alt="截屏2024-10-18 00 52 36" src="https://github.com/user-attachments/assets/217e5fa9-1780-4954-b5fb-75a439316a93">

**Comment by dbkest** [comment]
Oh god! So it's the fault of the prior loss.

**Comment by shivammehta25** [comment]
Sorry for the delayed reponse I was away from the keyboard. I am glad it is resolved, just to confirm that you are aware of the [extra blank token addition](https://github.com/shivammehta25/Matcha-TTS/wiki/Why-do-we-add-a-blank-token-between-phones%3F).

**Comment by dbkest** [comment]
> Sorry for the delayed reponse I was away from the keyboard. I am glad it is resolved, just to confirm that you are aware of the [extra blank token addition](https://github.com/shivammehta25/Matcha-TTS/wiki/Why-do-we-add-a-blank-token-between-phones%3F).

So sorry for forgetting to reply. After reading the literature you recommended, I have some understanding of the blank state. Its function seems to be that for the same phoneme, it can absorb different pronunciation variations, such as pronunciation duration. Theoretically, prosody should be better than pre-forced alignment methods like MFA, because for the same phoneme with different pronunciations, there are no additional states to absorb them.
